src/App.jsx

```diff
@@ -24,18 +23,18 @@ import AppRoutes from './AppRoutes';
  */
 
 const App = () => (
-  <Router>
-    <SessionProvider>
-      <CssBaseline />
-      <ThemeProvider theme={theme}>
-        <Layout ariaLabel="Home Page">
-          <UserDataContextProvider>
+  <SessionProvider restorePreviousSession>
+    <CssBaseline />
+    <ThemeProvider theme={theme}>
+      <UserDataContextProvider>
+        <BrowserRouter>
```

Comment:
Just a bit curious, is there a reason for moving the Router from the uppermost later to just before Layout compared to before where the Router is wrapping around everything?

Reply:
No strong reason, just thought it looked cleaner. Based the idea off of bulletproof react:

https://github.com/alan2207/bulletproof-react/tree/master

Which just contains a lot of good react programming patterns, and does the routes and provider structure that way.